src/sheet/wire-item.c

```diff
@@ -263,8 +262,6 @@ WireItem *wire_item_new (Sheet *sheet, Wire *wire)
 	item_data->changed_handler_id = g_signal_connect_object (
 	    G_OBJECT (wire), "changed", G_CALLBACK (wire_changed_callback), G_OBJECT (wire_item), 0);
 
-	g_signal_connect (wire, "delete", G_CALLBACK (wire_delete_callback), wire_item);
-
```

Comment:
so how exactly are wire deletions now going to be handled?

Reply:
Hello Bernhard,

the callback function that I removed was attempting to remove the GooCanvas item, but I think this is removed by a similar function call from sheet_delete_selection()...

What do you say ?

Reply:
Did you test this thoroughly?

Reply:
I have tested by selecting a wire and removing it. What other tests do you suggest ?

Reply:
Selecting multiple wires. Selecting wires with intersection nodes. Deleting wire by keyboard (if that ever worked..).

Reply:
- Selecting multiple wires and deleting them: works.
- Deleting wires by keyboard: works.
- Selecting wires with intersection nodes: works.

What else ?

Reply:
The only real difference with wires that I cannot understand yet is the following one:

--- /home/guido/github.com/oregano/src/sheet/sheet.c	2017-09-13 18:23:19.140711404 +0200
+++ src/sheet/sheet.c	2017-09-13 21:48:31.075661070 +0200
@@ -1026,6 +1026,7 @@ void sheet_delete_selection (Sheet *shee
 	for (iter = copy; iter; iter = iter->next) {
 		sheet_remove_item_in_sheet (SHEET_ITEM (iter->data), sheet);
 		goo_canvas_item_remove (GOO_CANVAS_ITEM (iter->data));
+		g_object_unref (iter->data); // crashes with wires only !
 	}
 	g_list_free (copy);

Can you shed some light on the above ? Thanks !

Reply:
I'll try to dig into it this weekend.

Reply:
The above change is not in the pull request, so for the time being you can merge the latter, unless there are other open issues....

Reply:
For your information, the stack trace when trying to g_object_unref (iter->data) with wires is as follows:

#0  0x00007ffff50a59d2 in goo_canvas_item_set_parent () at /usr/lib64/libgoocanvas-2.0.so.9
#1  0x00007ffff50a1d38 in goo_canvas_group_dispose () at /usr/lib64/libgoocanvas-2.0.so.9
#2  0x00007ffff748de73 in g_object_unref () at /usr/lib64/libgobject-2.0.so.0
#3  0x000000000043da0e in sheet_delete_selection ()

Reply:
Ok, it's fixed here now:

https://github.com/gtrentalancia/oregano/commit/f3cfd251a348e9349880eb74d4b1b406f900cd7e

I think you can merge everything and, if you like, increase the version number...

Thanks.

Reply:
After fixing a major bug in the part item and sheet item code (reference leak), I have bumped the version...